plugins/uefi/fu-uefi-common.h

```diff
@@ -57,6 +57,9 @@ typedef struct __attribute__((__packed__)) {
 	guint32		 status;
 } efi_update_info_t;
 
+/* the biggest size SPI part currently seen */
+#define FU_UEFI_COMMON_REQUIRED_ESP_FREE_SPACE		(32 * 1024 * 1024)
```

Comment:
I recall we talked about making this biggest known SPIx2 for recovery image in HDD. Although that might be Overkill since an "old" recovery image should already be there unless manually deleted on this boot.

So in summary this seems fine.